x/incentives/keeper/msg_server.go

```diff
@@ -63,10 +66,20 @@ func (server msgServer) AddToGauge(goCtx context.Context, msg *types.MsgAddToGau
 		return nil, err
 	}
 
-	if err := server.keeper.chargeFeeIfSufficientFeeDenomBalance(ctx, owner, types.AddToGaugeFee, msg.Rewards); err != nil {
+	gauge, err := server.keeper.GetGaugeByID(ctx, msg.GaugeId)
+	if err != nil {
+		return nil, err
+	}
+
+	// Charge fess based on the number of coins to add
+	// Fee = AddToGaugeBaseFee + AddDenomFee * (NumAddedDenoms + NumGaugeDenoms)
```

Comment:
Why do we need to pay for existing denoms as well? Isn't it already paid for when added initially?